network/h2c.go

```diff
@@ -59,13 +59,11 @@ func newH2CTransport(disableCompression bool) http.RoundTripper {
 
 // newH2Transport constructs a neew H2 transport. That transport will handles HTTPS traffic
 // with TLS config.
-func newH2Transport(disableCompression bool, tlsConf *tls.Config) http.RoundTripper {
+func newH2Transport(disableCompression bool, tlsContext DialTLSContextFunc) http.RoundTripper {
 	return &http2.Transport{
 		DisableCompression: disableCompression,
-		DialTLS: func(netw, addr string, tlsConf *tls.Config) (net.Conn, error) {
-			return DialTLSWithBackOff(context.Background(),
-				netw, addr, tlsConf)
+		DialTLSContext: func(ctx context.Context, network, addr string, cfg *tls.Config) (net.Conn, error) {
+			return tlsContext(ctx, network, addr)
```

Comment:
Any reason for dropping `DialTLSWithBackOff` ?

Reply:
`DialWithBackOff` is called downstream https://github.com/knative/serving/pull/14452.
If we keep `DialWithBackOff` and call it here, we need to bring `tlsConf`  as an extra argument of `newH2Transport()`. Is it better to call `DialTLSWithBackOff` here?